app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationManager.kt

```diff
@@ -0,0 +1,98 @@
+package com.randomappsinc.studentpicker.presentation
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.content.res.AssetFileDescriptor
+import android.media.AudioManager
+import android.media.MediaPlayer
+import android.os.Handler
+import android.view.View
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.DecelerateInterpolator
+import com.randomappsinc.studentpicker.R
+import com.randomappsinc.studentpicker.utils.UIUtils
+
+/** The amount of time you need to wait for the drumroll crash to happen*/
+private const val ANIMATION_DELAY_MS = 2600L
+
+class PresentationManager(
+        private val listener: Listener,
+        private val view: View) {
+
+    interface Listener {
+        fun speakNames()
+    }
+
+    private var player: MediaPlayer = MediaPlayer()
+    private var handler: Handler = Handler()
+    private val animateNamesTask = ::animateNames
+    private var defaultAnimationLengthMs: Long = view.context.resources.getInteger(
```

Comment:
Isn't it clearly an integer? We don't need values greater than 2 billion.
```suggestion
    private var defaultAnimationLengthMs: Integer = view.context.resources.getInteger(
```

Reply:
The ObjectAnimator duration is taking long, Kotlin is weird it doesn't have an implicit casting like java I think that because it doesn't have a primitive data type.